bottlerocket/agents/src/workload.rs

```diff
@@ -0,0 +1,155 @@
+use crate::error;
+use crate::sonobuoy::process_sonobuoy_test_results;
+use bottlerocket_types::agent_config::{WorkloadConfig, WORKLOAD_RESULTS_FILENAME};
+use log::{info, trace};
+use model::TestResults;
+use snafu::{ensure, ResultExt};
+use std::fs::File;
+use std::io::Write;
+use std::path::Path;
+use std::process::Command;
+
+const SONOBUOY_BIN_PATH: &str = "/usr/bin/sonobuoy";
+
+/// Runs the workload conformance tests according to the provided configuration and returns a test
+/// result at the end.
+pub async fn run_workload(
+    kubeconfig_path: &str,
+    workload_config: &WorkloadConfig,
+    results_dir: &Path,
+) -> Result<TestResults, error::Error> {
+    info!("Processing workload test plugins");
+    let mut plugin_test_args: Vec<String> = Vec::new();
+    for (id, plugin) in workload_config.plugins.iter().enumerate() {
+        info!("Initializing test {}-{}", id, plugin.name);
+        let plugin_yaml = format!("{}-plugin.yaml", plugin.name);
+        let output = Command::new(SONOBUOY_BIN_PATH)
+            .arg("gen")
+            .arg("plugin")
+            .arg("--name")
+            .arg(plugin.name.clone())
+            .arg("--image")
+            .arg(plugin.image.clone())
+            .output()
+            .context(error::WorkloadProcessSnafu)?;
+        ensure!(
+            output.status.success(),
+            error::WorkloadPluginSnafu {
+                plugin: plugin.name.clone()
+            }
+        );
+
+        // Write out the output to a file we can reference later
+        let mut f = File::create(plugin_yaml.clone()).context(error::WorkloadProcessSnafu)?;
+        f.write_all(&output.stdout)
+            .context(error::WorkloadProcessSnafu)?;
+
+        // Add plugin to the arguments to be passed to sonobuoy run
+        plugin_test_args.append(&mut vec!["--plugin".to_string(), plugin_yaml.to_string()]);
```

Comment:
Because I suggested making it a path...

```suggestion
        plugin_test_args.append(&mut vec!["--plugin".to_string(), plugin_yaml.display().to_string()]);
```